File: Leetcode/14-longest-common-prefix.js
const longestCommonPrefix = function(strs)
{
    let commonPrefix = '';

    for(let i = 0; i < strs[0].length; i++)
    {
        let temp = strs[0][i];
        for(let j = 0; j < strs.length; j++)
        {
           if(strs[j][i] !== temp)
                return commonPrefix;
        }
        commonPrefix += temp;
    }
    return commonPrefix;
}

console.log(longestCommonPrefix(["dog","racecar","car"]));

File: Leetcode/14-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        common_prefix = ''
        
        for i in range(len(strs[0])):
            current_char = strs[0][i]
            for j in range(len(strs)):
                if i == len(strs[j]) or strs[j][i] != current_char:
                    return common_prefix
            
            common_prefix += current_char
        
        return common_prefix